src/compute-client/src/controller/instance.rs

```diff
@@ -521,6 +521,29 @@ impl<T: ComputeControllerTimestamp> Instance<T> {
         );
     }
 
+    /// Returns `true` iff all collections are hydrated on at least one replica.
+    pub fn check_any_replica_hydrated(&self) -> bool {
+        let mut any_hydrated = false;
+        for (replica_id, replica_state) in self.replicas.iter() {
+            let mut replica_hydrated = true;
+
+            for collection in replica_state.collections.values() {
+                if !collection.hydrated() {
+                    replica_hydrated = false;
+
+                    tracing::info!("replica {replica_id} is not hydrated");
+                    break;
+                }
+            }
+
+            if replica_hydrated {
+                any_hydrated = true;
+            }
```

Comment:
Could you also add a comment here that continuing the loop is intentional so we log about all not hydrated replicas?